### DISCUSSION TITLE
how to use Powershell in runtime:6.0

### DISCUSSION THREAD

**Discussion by shengzhizhou** [dotnet/dotnet-docker#3930] [Q&A] [answered]
Hi, I'm working on a new .NET 6 project and I need to call some powershell scripts in program. I notice runtime:6.0 doesn't have PowerShell built-in. I know sdk:6.0 has built-in PowerShell but the image size will be much larger as final image. What else can I do? How can I achieve that? Thanks for answers!

**Comment by mthalman** [comment, marked as answer]
You can grab it out of the sdk image like this:

### Linux

```Dockerfile
FROM mcr.microsoft.com/dotnet/runtime:6.0

COPY --from=mcr.microsoft.com/dotnet/sdk:6.0 /usr/share/powershell /usr/share/powershell
RUN ln -s /usr/share/powershell/pwsh /usr/bin/pwsh
```

### Windows

```Dockerfile
FROM mcr.microsoft.com/dotnet/runtime:6.0

COPY --from=mcr.microsoft.com/dotnet/sdk:6.0 [ "/Program Files/powershell", "/Program Files/powershell" ]
COPY --from=mcr.microsoft.com/dotnet/sdk:6.0 [ "/Program Files/dotnet/shared/Microsoft.WindowsDesktop.App", "/Program Files/dotnet/shared/Microsoft.WindowsDesktop.App" ]

# In order to set system PATH, ContainerAdministrator must be used
USER ContainerAdministrator
RUN setx /M PATH "%PATH%;C:\Program Files\powershell"
USER ContainerUser
```

**Reply by mthalman** [reply]
Updated my original answer to include solutions for both Linux and Windows.

**Reply by rcdailey** [reply]
Even when I use `dotnet/sdk:6.0-alpine`, I cannot use `pwsh` (from bash, it says it cannot be found). Is the `ln -s` necessary in this case as well?

**Reply by mthalman** [reply]
> Even when I use `dotnet/sdk:6.0-alpine`, I cannot use `pwsh` (from bash, it says it cannot be found). Is the `ln -s` necessary in this case as well?

No changes are necessary to use PowerShell in the SDK image. It's built into that image.
```
C:\> docker run --rm mcr.microsoft.com/dotnet/sdk:6.0-alpine pwsh
Unable to find image 'mcr.microsoft.com/dotnet/sdk:6.0-alpine' locally
6.0-alpine: Pulling from dotnet/sdk
530afca65e2e: Already exists
6207422b9602: Already exists
97f3adc41521: Already exists
5b011df4216c: Already exists
4f47c84b20eb: Already exists
c98422e87e60: Already exists
d45206748b5a: Already exists
Digest: sha256:e0575d5756d32a4679f0cdec32a6fac22cbd43352ffe8e52c1620a34d9386a9d
Status: Downloaded newer image for mcr.microsoft.com/dotnet/sdk:6.0-alpine
PowerShell 7.2.5
Copyright (c) Microsoft Corporation.

https://aka.ms/powershell
Type 'help' to get help.

PS />
```

If you're having issues, I would suggest creating a different discussion post to troubleshoot. Please provide details in that case.
